content/2016.02.localisation-service.md

```diff
@@ -0,0 +1,316 @@
+
+```javascript
+    var gLocalizedStrings = new Map();
+
+    var MozLoopServiceInternal = {
+      // ...
+
+      configureGLocalizedStringsWithKinto: function() {
+        let FirefoxAdapter = Kinto.adapters.FirefoxAdapter;
+        let config = {
+           remote: "https://kinto-ota.dev.mozaws.net/v1",
+           bucket: "loop-client",
+           adapter: FirefoxAdapter
+        };
+        let db = new Kinto(config);
+        let collectionName = "fr"; // XXX: this.locale getter can help
+        let locales = db.collection(collectionName);
+  
+        return locales.db.open()
+          .then(() => {
+            return locales.sync();
+          })
+          .then(() => {
+            return locales.list();
+          })
+          .then((res) => {
+            res.data.forEach((record) => {
+              gLocalizedStrings.set(record.key, record.value);
+            });
+          })
+          .then(() => {
+            return locales.db.close();
+          });
+        },
+
+        initialize: Task.async(function*() {
+          // ...
+
+          // Start l10n initilization
+          yield this.configureGLocalizedStringsWithKinto();
+
+          // ...
+        },
+      // ...
+    }
+```
+
+## Demo
+
+{% video https://kinto-ota.dev.mozaws.net/attachments/demo.mp4 900 550 %}
+
+
+# Conclusion
+
+One of our objectives with this experiment was to illustrate what can
+be done with Kinto within Firefox.
+
+We have seen that using the embedded Kinto.js in Firefox is very
+simple, and playful.
+
+Relying on JSON schema to validate the records on the server also
+proved useful for generating web forms in Kinto-admin. A Web UI really
+helps to interact with the data, and the current product is only first
+step!
+
+With this article, we hope that our readers find inspiration and
+imagine totally different use-cases!
+
+
+# Going further?
+
+The localization was a proof-of-concept that looks promising.
+
+If it would have to be done seriously and ship in production, there
+are still a number of blockers that should be investigated with the
+help of the localization team.
+
+
+## XUL and DTD Entities file loaded from Kinto
+
+This experiment is based on the Firefox Hello component which is using
+a properties file with the translations for each different view.
+
+Other components are using properties files but some of them are also
+using DTD Entity files for translation.
+
+We would need to investigate how we can load the Entities file
+dynamically.
+
+
+## Tooling / Workflow
+
+If we were to use Kinto we would need to integrate in the existing
+translation flow.
+
+This means letting [Elmo release managers](https://l10n.mozilla.org/) to ship
+translation in the right Kinto collection for instance.
+
+
+## l10n vs l20n: Asynchroneous translation loading
+
+Today it seems that the translation loading is synchroneous into Firefox.
+
+It could be interesting to migrate from l10n.js to l20n.js to load
```

Comment:
it might make sense to mention that l10n.js isn't supported anymore;  I wouldn't like people to go on and start using it today.